op-challenger/game/fault/trace/alphabet/provider.go

```diff
@@ -50,12 +50,13 @@ func (ap *AlphabetTraceProvider) GetStepData(ctx context.Context, pos types.Posi
 		return absolutePrestate, []byte{}, preimageData, nil
 	}
 	// We want the pre-state which is the value prior to the one requested
-	prestateTraceIndex := posIndex.Sub(posIndex, big.NewInt(1))
+	prestateTraceIndex := new(big.Int).Sub(posIndex, big.NewInt(1))
 	if prestateTraceIndex.Cmp(new(big.Int).SetUint64(ap.maxLen)) >= 0 {
 		return nil, nil, nil, fmt.Errorf("%w depth: %v index: %v max: %v", ErrIndexTooLarge, ap.depth, posIndex, ap.maxLen)
 	}
-	claim := BuildAlphabetPreimage(big.NewInt(0), absolutePrestateInt)
-	for i := big.NewInt(0); i.Cmp(posIndex) < 0; i = i.Add(i, big.NewInt(1)) {
+	// First step expands the absolute preimage to its full form. Weird but it's how AlphabetVM works.
+	claim := ap.step(absolutePrestate)
+	for i := big.NewInt(0); i.Cmp(prestateTraceIndex) <= 0; i = i.Add(i, big.NewInt(1)) {
 		claim = ap.step(claim)
 	}
 	return claim, []byte{}, preimageData, nil
```

Comment:
We should actually just be able to use the `posIndex` again here - I think maybe it was changed in this pr since the prior `prestateTraceIndex` calculation subtracted one from the `posIndex` by mistake. ie, the following should still work
```
for i := big.NewInt(0); i.Cmp(posIndex) < 0; i = i.Add(i, big.NewInt(1)) {
```

Reply:
I found it very confusing that we calculated the step prior to the trace index and then didn't actually use it for the loop. So I think we're better off doing the subtraction and then consistently using the value (rabbit be damned). Probably could go either way to be honest - if we remove the iteration it will remove any chance of confusion here too I suspect.